### PR TITLE
chore: standardize status api errors for task config policies

### PR DESCRIPTION
**Pull request by carolinaecalderon**
## Ticket
CM-556

## Description
Some of the APIs return a status, e.g. status.Errorf(codes.InvalidArgument, errMessage) and some return a simple error, e.g. fmt.Errorf("error parsing config policies: %w", err). Review the task config policy APIs and ensure we are returning a status code in exported/able functions.

## Test Plan
N/A -- make sure e2e tests pass

## Checklist

- [ ] Changes have been manually QA'd
- [ ] New features have been approved by the corresponding PM
- [ ] User-facing API changes have the "User-facing API Change" label
- [ ] Release notes have been added as a separate file under `docs/release-notes/`
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses have been included for new code which was copied and/or modified from any external code

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: master/internal/api_config_policies.go [hunk 1]
```diff
@@ -149,13 +150,13 @@ func (a *apiServer) PutWorkspaceConfigPolicies(
 		Constraints:     constraints,
 	})
 	if err != nil {
-		return nil, fmt.Errorf("error setting task config policies: %w", err)
+		return nil, status.Errorf(codes.Internal, settingPoliciesErr+err.Error())
```
Comment (amandavialva01) [inline master/internal/api_config_policies.go:153]: do we need a colon after `settingPoliciesErr`? Or is this taken care of with the string concat?

File: master/internal/api_config_policies.go [hunk 2]
```diff
@@ -56,15 +57,15 @@ func (a *apiServer) validatePoliciesAndWorkloadType(
...
+		return fmt.Errorf(invalidWorkloadTypeErr+": %s", workloadType)
 	}
 }
 
 func parseConfigPolicies(configAndConstraints string) (
 	tcps map[string]interface{}, invariantConfig *string, constraints *string, err error,
 ) {
 	if len(configAndConstraints) == 0 {
-		return nil, nil, nil, status.Error(codes.InvalidArgument, "nothing to parse, empty "+
+		return nil, nil, nil, fmt.Errorf("nothing to parse, empty " +
```
Comment (amandavialva01) [inline master/internal/api_config_policies.go:68]: nice!

File: master/internal/api_config_policies.go [hunk 3]
```diff
@@ -56,15 +57,15 @@ func (a *apiServer) validatePoliciesAndWorkloadType(
 		return configpolicy.ValidateNTSCConfig(globalConfigPolicies, configPolicies,
 			priorityEnabledErr)
 	default:
-		return status.Errorf(codes.InvalidArgument, fmt.Sprintf(invalidWorkloadTypeErr+": %s.", workloadType))
+		return fmt.Errorf(invalidWorkloadTypeErr+": %s", workloadType)
```
Comment (amandavialva01) [inline master/internal/api_config_policies.go:60]: isn't the workload type part of the request?
I think we may wanna leave this as `status.InvalidArgument`

File: master/internal/api_config_policies.go [hunk 4]
```diff
@@ -34,16 +35,16 @@ func (a *apiServer) validatePoliciesAndWorkloadType(
...
 	// Validate workload type
 	if !configpolicy.ValidWorkloadType(workloadType) {
-		errMessage := fmt.Sprintf(invalidWorkloadTypeErr+": %s.", workloadType)
+		errMessage := fmt.Errorf(invalidWorkloadTypeErr+": %s", workloadType)
 		if len(workloadType) == 0 {
-			errMessage = noWorkloadErr
+			errMessage = fmt.Errorf(noWorkloadErr)
 		}
-		return status.Errorf(codes.InvalidArgument, errMessage)
+		return errMessage
```
Comment (amandavialva01) [inline master/internal/api_config_policies.go:42]: [ditto](https://github.com/determined-ai/determined/pull/10119/files#r1815246879)

File: master/internal/api_config_policies.go (hunk 1 above)
Comment (carolinaecalderon) [reply master/internal/api_config_policies.go:153]: done

File: master/internal/api_config_policies.go (hunk 3 above)
Comment (carolinaecalderon) [reply master/internal/api_config_policies.go:60]: Everywhere a `validatePoliciesAndWorkloadType` err is reference handles it like
```
err = a.validatePoliciesAndWorkloadType(globalConfigPolicies, req.WorkloadType,
		req.ConfigPolicies)
	if err != nil {
		return nil, status.Errorf(codes.InvalidArgument, err.Error())
	}
```
so `InvalidArgument` is there!

Comment (amandavialva01) [reply master/internal/api_config_policies.go:60]: Ooh gotcha!! Great catch thanks Caroina!

**Review by amandavialva01 — APPROVED** [review]
Looks great, awesome work!

**Review by kkunapuli — APPROVED** [review]

**Pull request merged**